src/entity/object/FallingBlock.php

```diff
@@ -146,6 +150,21 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 		return $hasUpdate;
 	}
 
+	protected function onHitGround() : ?float{
+		if($this->block instanceof Fallable){
+			$damagePerBlock = $this->block->getFallDamagePerBlock();
+			if($damagePerBlock > 0 && ($fallenBlocks = round($this->fallDistance) - 1) > 0){
+				foreach($this->getWorld()->getCollidingEntities($this->getBoundingBox()) as $entity){
+					if($entity instanceof Living){
+						$ev = new EntityDamageByEntityEvent($this, $entity, EntityDamageEvent::CAUSE_FALLING_BLOCK, min($fallenBlocks * $damagePerBlock, $this->block->getMaxFallDamage()));
```

Comment:
I think the damage amount can be extracted into a variable instead of recalculating it every iteration.